packages/styled-system/src/style-config.ts

```diff
@@ -66,15 +61,16 @@ function createResolver(theme: Theme) {
 
       if (isMultipart) {
         config.parts?.forEach((part) => {
-          mergeWith(result, {
+          compact({
+            ...result,
             [part]: isSingle ? styles[part] : { [query]: styles[part] },
           })
         })
         continue
       }
 
       if (!isMultipart) {
-        if (isSingle) mergeWith(result, styles)
+        if (isSingle) compact({ ...result, ...styles })
```

Comment:
This is wrong because mergeWith is a deep merge and what you have here is a shallow merge. Please revert to mergeWith.